src/k8s/pkg/k8sd/features/calico/network.go

```diff
@@ -65,10 +72,15 @@ func ApplyNetwork(ctx context.Context, snap snap.Snap, cfg types.Network, _ type
 		},
 		"installation": map[string]any{
 			"calicoNetwork": map[string]any{
-				"ipPools": podIpPools,
+				"ipPools":                    podIpPools,
+				"nodeAddressAutodetectionV4": config.autodetectionV4,
+				"nodeAddressAutodetectionV6": config.autodetectionV6,
```

Comment:
careful as helm values still need to be `map[string]any` at the end of the day. perhaps `mapstructure` can help us here.